### PR TITLE
fix: validate files to be uploaded

### PR DESCRIPTION
**Pull request by trevorblades**
This PR is part one of two, ensuring that only files of specific types are uploaded to our system. This does file upload validation on the client side, but we must also implement some server-side validation.

Fixes FER-2070

### CODE REVIEW

**Review by dsinghvi — COMMENTED** [review]
@trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))? 

This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.

**Comment by trevorblades** [comment]
> @trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))?
> 
> This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.

That's a good idea! I can do that. Does the docs validator run before docs publishing, and would its failure stop a publish command? What I'm getting at is if you see this as something that lives in the docs validator _and_ here in the publish command, or just in the validator.

**Comment by dsinghvi** [comment]
> > @trevorblades thoughts on adding this to the `docs-validation` package ([link](https://github.com/fern-api/fern/tree/main/packages/cli/yaml/docs-validator/src/rules))?
> > This way when a user runs `fern check` they will get a validation error that will automatically notify them that they can't upload those specific files.
> 
> That's a good idea! I can do that. Does the docs validator run before docs publishing, and would its failure stop a publish command? What I'm getting at is if you see this as something that lives in the docs validator _and_ here in the publish command, or just in the validator.

Just in the validator works! the validator runs and has to be passed in order to hit publish

**Comment by trevorblades** [comment]
@dsinghvi I moved the file type check to the validator CLI and added some tests to confirm it's working as expected. Lmk if that looks like what you had in mind!

**Review by dsinghvi — APPROVED** [review]
🚀 thank you!

File: packages/cli/yaml/docs-validator/src/rules/valid-file-types/valid-file-types.ts [hunk 1]
```diff
@@ -27,16 +28,20 @@ export const ValidFileTypes: Rule = {
...
+
+                    if (!isValid) {
+                        return [
+                            {
+                                severity: "error",
+                                message: `File type of ${value} is invalid`
+                            }
+                        ];
+                    }
                 }
```
Comment (abvthecity) [inline packages/cli/yaml/docs-validator/src/rules/valid-file-types/valid-file-types.ts:45]: should we add a warning or error if the file doesn't exist?

Comment (dsinghvi) [reply packages/cli/yaml/docs-validator/src/rules/valid-file-types/valid-file-types.ts:45]: there's another rule that checks that!

**Pull request merged**